cookbooks/aws-parallelcluster-environment/recipes/init/mount_internal_use_fs.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+return if on_docker?
+efs_shared_dir_array = node['cluster']['efs_shared_dirs'].split(',')
+efs_fs_id_array = node['cluster']['efs_fs_ids'].split(',')
+efs_encryption_in_transit_array = node['cluster']['efs_encryption_in_transits'].split(',')
+efs_iam_authorization_array = node['cluster']['efs_iam_authorizations'].split(',')
+
+internal_shared_dir_array = []
+internal_efs_fs_id_array = []
+internal_efs_encryption_array = []
+internal_efs_iam_array = []
+
+# Identify the internal use filesystems and store their data in arrays for the EFS resource
+efs_shared_dir_array.each_with_index do |dir, index|
+  next unless node['cluster']['internal_shared_dirs'].include?(dir)
+  # Don't mount the login nodes shared dir to compute nodes
+  next if node['cluster']['node_type'] == 'ComputeFleet' && dir == node['cluster']['shared_dir_login_nodes']
+  internal_shared_dir_array.push(dir)
+  internal_efs_fs_id_array.push(efs_fs_id_array[index])
+  internal_efs_encryption_array.push(efs_encryption_in_transit_array[index])
+  internal_efs_iam_array.push(efs_iam_authorization_array[index])
+end
```

Comment:
I'm not sure.  I prefer having code here that just filters the ones we need for the init and a separate one that filters what we need for the cx.  We do this filtering process twice, so I wonder if it might be better to just put this into the attributes file and then leverage the attributes we need in each recipe.  This did get auto merged with Giacomo's approval, but I will do this refactoring as a follow up